lessons/misc/intro-bash/lesson.md

```diff
@@ -0,0 +1,354 @@
+ls -l
+```
+
+You can also string multiple flags together. Below, we tell the computer to *list* all (`-h`) with all details (`-l`, e.g. size, date, permissions...), with all sizes in human-readable format (`-h`) and to sort by filesize (`-S`)
+
+```
+ls -alhS
+# Equivalent to ls -a -l -h -S
+```
+
+It's worth keeping in mind that nearly every shell command has its own set of flags/modifiers. We'll be going over the more generally useful ones as they come. If you want to see how to use a particular command along with its available flags, you can use the **man**ual command.
+
+```
+man ls
+```
+
+## Working with directories
+
+Back to our folder here. While peeking at the contents of a directory is certainly useful, we'll often find ourselves wanting to switch our current directory. This is done using the `cd` (**c**hange **d**irectory) command. You should have downloaded a folder call `intro-bash-demo` and saved it somewhere practical (e.g. your Desktop or home folder) - let's use `cd` to navigate to it. I've a included a snippet of some useful `cd` commands below:
+
+* `cd <directory name>`: Change ito directory. Relative to your current working directory.
+    * `cd intro-bash-demo/`: Change into `intro-bash-demo directory`
+    * `cd ../`: Go back one directory from the previous one. Use `../../` to go back 2 directories.
+    * `cd ~/`: Change to your home directory
+
+**Tip:** You can press TAB to atuocomplete directory and filenames. If a file in the working directory matches the start of the file or directory you've started typing, the shell with autocomplete its name.
+
+Once you've navigated to the `intro-bash-demo` folder, *list* the fles in the folder to view its contents.
+
+You notice some text files (e.g. animals.txt) and directories. We'll return to these shortly.
+
+We can create directories with the `mkdir` command. Let's create a directory called `text-files`.
+
+```
+mkdir text-files
+```
+
+Similarly, you can remove directories with the `rmdir` command. Let's remove the `folder-to-delete` folder.
+
+```
+rmdir folder-to-delete
+```
+
+Hmmm...that didn't work. It turns out `rmdir` only works on empty directories. Let's make sure we don't actually need the files in that directory before trying again to delete it. You can pass directory names to `ls` to its contents.
+
+```
+ls folder-to-delete
+```
+
+Now that we're sure we don't need that folder anymore, let's **f**orcefully and **r**ecursively delete it using the `rm` (for *remove*) command. Note, this command will not warn you and the content will be gone forever.
+
+```
+rm -rf folder-to-delete
+```
+
+`rm` can also be used on files without the `-rf` flags. Try remove `to-delete.txt`
+
+## Working with files
+
+Beyond navigating directories, bash offers a suite of useful tools to do all sorts of things with files themselves.
+
+Let's examine `hello_world.py` in greater detail. We can do this with `cat`, which simply spits out the contents of a file to the terminal's output.
```

Comment:
Not needed, but maybe something about how this is a Python script and that we can run these from the bash shell? I know we cover this later, but it might be good to prime that knowledge here and show that a Python script is really just a plain text file at the end of the day